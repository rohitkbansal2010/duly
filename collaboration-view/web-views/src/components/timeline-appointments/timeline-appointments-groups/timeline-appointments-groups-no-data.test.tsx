import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { calendarTwoBlueIcon } from '@icons';

import {
  TimelineApppointmentsGroupsNoData,
  TimelineApppointmentsGroupsNoDataPropsType
} from './timeline-appointments-groups-no-data';

const mockProps: TimelineApppointmentsGroupsNoDataPropsType = {
  icon: calendarTwoBlueIcon,
  backgroundIcon: 'Blue',
  title: 'Recent Appointments',
  colorTitle: 'LightBlue',
};

describe('TimelineApppointmentsGroupsNoData', () => {
  let component: ReturnType<typeof shallow>;

  beforeEach(() => {
    component = shallow(<TimelineApppointmentsGroupsNoData {...mockProps} />);
  });

  it('should render the TimelineApppointmentsGroupsNoData component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render icon', () => {
    const iconElement = component.find('[data-test="appointment-icon"]');

    const expectedProps = {
      src: mockProps.icon,
      alt: 'Calendar icon',
      'data-test': 'appointment-icon',
    };

    expect(iconElement).toHaveLength(1);
    expect(iconElement.props()).toEqual(expectedProps);
  });

  it('should render title', () => {
    const titleElement = component.find('[data-test="appointment-title"]');

    const expectedTitle = `No ${mockProps.title}`;

    expect(titleElement).toHaveLength(1);
    expect(titleElement.text().includes(expectedTitle)).toBeTruthy();
  });
});
