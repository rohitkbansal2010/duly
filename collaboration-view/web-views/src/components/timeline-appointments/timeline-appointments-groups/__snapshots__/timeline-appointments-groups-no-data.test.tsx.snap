// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineApppointmentsGroupsNoData should render the TimelineApppointmentsGroupsNoData component 1`] = `
<div
  className="timelineAppointmentsGroupsNoData"
>
  <div
    className="timelineAppointmentsGroupsNoDataIcon timelineAppointmentsGroupsNoDataIconBlue"
  >
    <img
      alt="Calendar icon"
      data-test="appointment-icon"
      src="calendar-two-blue.svg"
    />
  </div>
  <div
    className="timelineAppointmentsGroupsNoDataTitle timelineAppointmentsGroupsNoDataTitleLightBlue"
    data-test="appointment-title"
  >
    No Recent Appointments
  </div>
</div>
`;
